src/main/java/org/jenkinsci/plugins/codesonar/CodeSonarPublisher.java

```diff
@@ -414,15 +396,26 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
                     .filter(c -> c.getProjectName() != null && c.getProjectName().equals(expandedProjectName))
                     .collect(Collectors.toList());
             if(filteredActions != null && !filteredActions.isEmpty() && filteredActions.size() < 2) {
-                csLogger.writeInfo("Found comparison data");
+                csLogger.writeInfo("Found comparison build data");
                 compareDTO = filteredActions.get(0).getBuildActionDTO();
             }
         }
+
+        CodeSonarHubAnalysisDataLoader previousDataLoader = null;
+        if(compareDTO != null) {
+            previousDataLoader = new CodeSonarHubAnalysisDataLoader(httpService, hubInfo, compareDTO.getBaseHubUri(), compareDTO.getAnalysisId(), visibilityFilter, newWarningsFilter);
+            csLogger.writeInfo("Comparison analysis: analysisId {0} from hub \"{1}\"", compareDTO.getAnalysisId(), compareDTO.getBaseHubUri());
+            //Signal on the Jenkins' console if there's a mismatch between the hub of the current analysis and the one of the comparison analysis
+            if(!currentDataLoader.getBaseHubUri().equals(previousDataLoader.getBaseHubUri())) {
+                csLogger.writeInfo("Warning: hub URI for previous build does not match current build. previous=\"{0}\", current=\"{1}\"", previousDataLoader.getBaseHubUri(), currentDataLoader.getBaseHubUri());
+                
+            }
+        }
         
         csLogger.writeInfo("Evaluating conditions");
 
         for (Condition condition : conditions) {
-            Result validationResult = condition.validate(buildActionDTO, compareDTO, launcher, listener, csLogger);
+            Result validationResult = condition.validate(currentDataLoader, previousDataLoader, null, null, launcher, listener, csLogger);
```

Comment:
We should remove the nulls from this `Condition.validate()` call by removing the formal parameters from the `validate()` method.